File: Philosophers_java/src/Table.java
import java.util.concurrent.Semaphore;

public class Table {
    private Semaphore[] _forks;

    public Table(Integer forksCount){
        _forks = new Semaphore[forksCount];
        for (Integer i = 0; i < _forks.length; i++){
            _forks[i] = new Semaphore(1);
        }
    }

    public void acquireFork(Integer index){
        try {
            _forks[index].acquire();
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        }
    }

    public void releaseFork(Integer index){
        _forks[index].release();
    }
}

File: Philosophers_sharp/Philosophers_sharp/Table.cs
using System.Threading;

namespace Philosophers_sharp
{
    public class Table
    {
        private Semaphore[] _forks;

        public Table(int forksCount)
        {
            _forks = new Semaphore[forksCount];
            for (int i = 0; i < _forks.Length; i++)
            {
                _forks[i] = new Semaphore(1, 1);
            }
        }

        public void WaitOneFork(int index)
        {
            _forks[index].WaitOne();
        }

        public void ReleaseFork(int index)
        {
            _forks[index].Release();
        }
    }
}